projects/igniteui-angular/src/lib/grids/toolbar/grid-toolbar-advanced-filtering.component.ts

```diff
@@ -50,6 +50,26 @@ export class IgxGridToolbarAdvancedFilteringComponent implements AfterViewInit {
         this.grid?.advancedFilteringExpressionsTreeChange.subscribe(filteringTree => {
             this.numberOfColumns = this.extractUniqueFieldNamesFromFilterTree(filteringTree).length;
         });
+        this.grid?.filteringExpressionsTreeChange.subscribe(() => {
+            if (this.grid.advancedFilteringExpressionsTree) {
+                const dataKeys = Object.keys(this.grid.data[0]);
+                const filteringOperands = this.grid.advancedFilteringExpressionsTree.filteringOperands;
+
+                const allOperandsPresent = filteringOperands.every(operand => {
+                    if (operand instanceof FilteringExpressionsTree) {
+                        // Recursively check nested filtering trees
+                        return this.checkFilteringOperands(operand, dataKeys);
+                    } else {
+                        return dataKeys.includes(operand.fieldName);
+                    }
+                });
+
+                if (!allOperandsPresent) {
+                    this.grid.advancedFilteringExpressionsTree = this.grid.filteringExpressionsTree;
+                    this.numberOfColumns = 0;
+                }
```

Comment:
advanced filtering and excel style filtering are working independently in the grid -- so setting esf collection to advanced filtering collection causes some strange behavior .. please check the following recordings:
![advFilteringAndESF](https://github.com/user-attachments/assets/89e6eb2d-f71b-488f-b269-8744ab793314)

Reply:
On my end, everything works as expected as you can see from the next gif.

About setting esf collection (`filteringExpressionsTree`) to advanced filtering collection (`advancedFilteringExpressionsTree`).. setting this collection happens only when we change the grid data and therefore the filter expression tree is cleared and contains nothing.

{
 fieldName: undefined
 "filteringOperands": [],
 "operator": 0,
 "type": 0
}

Therefore I also clear the advance filtering tree by setting it to the empty default filtering expression tree which is the same thing when it is cleared. That's why here I set `numberOfColumns` to be equal to 0 because we cleared the advanced filters by setting it to an empty filtering tree.

I'm not sure if this is correct though and if something else should be thought of like null undefined or something else.

![1](https://github.com/user-attachments/assets/50fa30da-2652-4e13-b9ef-24743ac019a8)